src/mui/delete/Delete.js

```diff
@@ -135,11 +136,11 @@ Delete.propTypes = {
 function mapStateToProps(state, props) {
     return {
         id: decodeURIComponent(props.match.params.id),
-        data:
-            state.admin.resources[props.resource].data[
-                decodeURIComponent(props.match.params.id)
-            ],
-        isLoading: state.admin.loading > 0,
+        data: getResourceRecord(
```

Comment:
Alright, it seems to the job just fine, I was afraid that `redux-saga` selection synthax would cause a problem at first, but it seems it was made to handle  things this way. Only drawback I see at the moment is that it is a pain to find out wich props name are expected by higher order selectors, especially for the composite selectors.
I didn't refactor the specialized selectors (`mapStateToProps` functions on the components). It seems out of this scope from point of view, but it should be easily done now that there are all the generic selectors available.